pkg/networkmanager/interfaces.js

```diff
@@ -488,15 +471,20 @@ export function NetworkManagerModel() {
                 return def;
         }
 
-        function get_ip(first, addr_from_nm, route_from_nm, ip_to_text) {
+        function get_ip(first, ip_to_text) {
+            const dns_data = self.supports_dns_data
+                ? get(first, "dns-data", [])
+                : get(first, "dns", []).map(ip_to_text);
+
             return {
                 method: get(first, "method", "auto"),
                 ignore_auto_dns: get(first, "ignore-auto-dns", false),
                 ignore_auto_routes: get(first, "ignore-auto-routes", false),
-                addresses: get(first, "addresses", []).map(addr_from_nm),
-                dns: get(first, "dns", []).map(ip_to_text),
+                address_data: get(first, "address-data", []).map(ip_address_from_nm),
+                gateway: get(first, "gateway", first === "ipv4" ? "0.0.0.0" : "::"),
```

Comment:
I was surprised to see a new "Gateway" field in the UI to be pre-filled with "0.0.0.0" while all the other fields starts out empty:

<img width="839" height="314" alt="image" src="https://github.com/user-attachments/assets/524539f2-ad2b-4290-88e6-be1343ed24a8" />

Does that make sense?

Reply:
I see that originally the gateway was only "guessed" for ipv4 and it did not respect the prefix of subnet, I'll add the same capability back and as a followup I could extend it.

original code: `config.gateway = `${split[0]}.${split[1]}.${split[2]}.${split[3] === "1" ? "254" : "1"}`;`